internal/llm/provider/openai.go

```diff
@@ -367,6 +368,15 @@ func (o *openaiClient) stream(ctx context.Context, messages []message.Message, t
 						currentContent += choice.Delta.Content
 					} else if len(choice.Delta.ToolCalls) > 0 {
 						toolCall := choice.Delta.ToolCalls[0]
+						if strings.HasPrefix(toolCall.ID, "functions.") {
+							exID, ok := toolCallIDMap[toolCall.ID]
+							if !ok {
+								newID := uuid.NewString()
+								toolCallIDMap[toolCall.ID] = newID
+								toolCall.ID = newID
+							}
+							toolCall.ID = exID
```

Comment:
Looks like we have a logic error here. The assign from line 376 won't matter because you're always overriding on 378 anyway. You meant to have the second inside `else`?

Reply:
that was dumb wasn't it 😂